src/Hooks/useFetch.js

```diff
@@ -0,0 +1,40 @@
+import { useState, useEffect } from "react";
+
+function useFetch(url, options = {}) {
+	const [response, setResponse] = useState(null);
+	const [error, setError] = useState(null);
+	// What is a real use case for isLoading? It seems redundant to me.
+	const [isLoading, setIsLoading] = useState(false);
+
+	useEffect(() => {
+		if (options.abort !== true) {
+			(async () => {
+				setIsLoading(true);
+				try {
+					const response = await fetch(url);
+					if (response.ok) {
+						setIsLoading(false);

```

Comment:
I would move all the setIsLoading(false) declarations to a finally block.

Reply:
I ended up having to implement some cache handling in `useFetch` so I couldn't fit all the `setStatus` calls in the `finally` block. Maybe take another look at it.